Providers/Profile.swift

```diff
@@ -434,14 +439,28 @@ public class BrowserProfile: Profile {
          * Locking is managed by withSyncInputs. Make sure you take and release these
          * whenever you do anything Sync-ey.
          */
-        var syncLock = OSSpinLock()
+        var syncLock = OSSpinLock() {
+            didSet {
+                if oldValue != 0 && syncLock == 0 {
```

Comment:
No - just added it as a safety check in case we fire a double start event or something weird